doc/user-manual.md

```diff
@@ -0,0 +1,738 @@
+
+> With great power comes great responsibility
+
+Taskr recipes runs with the user privilege. If you are admin the recipe has all the possibilities the admin has.
+
+> To err is human, but for a real disaster you need a computer"
+
+What can harm a dhis2 by clicking the wrong option in a menu in the dataElement option, can turn into a disaster when the same operation is repeated in a loop in taskr. It can get worst when what is in the loop is not the correct action or was not meant to run on all data elements. Be careful.
+
+
+# Small tour
+
+## Open the app
+
+Once the app installed you can locate the app in the menu.
+
+![recipes-list](./user-manual-taskr-app.jpg)
+
+## Recipes list
+
+![recipes-list](./user-manual-recipes-list.jpg)
+
+
+The recipes list screen is accessible via the top left hamburger menu icon. A list of recipes is shown :
+   - grey out recipes are "built in".
+   - blank recipees are the one created and saved in your dhis2
+
+Click on the run button, you arrive fill on the recipe page.
+
+## Recipe page
+
+Fill in the parameters if needed, click run, the results will show under the buttons.
+
+![recipes-list](./user-manual-recipe-page.jpg)
+
+If an error occur in the recipe (syntax or dhis2 api calls), it will show up on the top of the screen.
+
+![recipes-list](./user-manual-error.jpg)
+
+If you are in the edit mode the recipe editor will show up.
+
+A recipe is composed of
+ - the code (javascript snippet)
+ - the paremeters definition (optional, json)
+ - the report (optional, markdown with idyll components)
+
+# Historical pains & motivations
+
+## 1. Turning an api call results to csv or a map was not easy
+
+Lets get all orgunits with their coordinates and ancestors as a csv. If the recipe return an array of things with a geometry/coordinates/coordinate field taskr will try to display it also as a map.
+
+```js
+const api = await dhis2.api();
+const ou = await api.get("organisationUnits", {
+  fields: "id,name,ancestors[id,name],geometry",
+  paging: false
+});
+
+return _.flattenObjects(ou.organisationUnits, ["geometry"]);
+```
+
+The table is filterable either via the search icon or more accurately via the filter dialog and you can download the data as a csv via the cloud icon.
+
+![orgunit-map](./user-manual-orgunit-table.jpg)
+
+The orgunits table is clickable, zoomable, you can switch between different layers. It supports simple coordinates or shapes as defined by dhis2 geometry/coordinates field.
+
+![orgunit-map](./user-manual-orgunit-map.jpg )
+
+## 2. Coding a dhis2 app is sometimes too much work for just a one shot specific task
+
+Let's say you want to rename all indicators starting by "INDX.Y name" to "IND X.Y - name".
+
+Setting up a dhis2 app requires time and developper knowledge. Taskr is a good tradeoff, you can install/update the app from the app store. With basic programming knowledge or using the standard recipes you can automate already a few tasks without the *initial setup cost*.
+
```

Comment:
This is more for me the alterlative to a locally run script. "Benefits of the recipe versus a script running on a developer machine is that it can be run again (and not only by a developer) and is deployed with the DHIS2 instance so will not be lost on a single person machine."